src/book/03-guides/04-general/01-git.mdx

```diff
@@ -0,0 +1,61 @@
+---
+section: Guides
+chapter: General
+title: Git
+description: Contributing to a NUbots project with git.
+slug: /guides/general/git
+---
+
+## Prerequisites
+
+This guide assumes you know a few basic shell commands, and the basics of git, such as adding, committing, and pushing to a repository.
+
+## Getting Code Into a Repository
+
+Let's suppose you have created a file which you would like to add to a NUbots repository called `MakeRobotStand.cpp`, which is part of your project to make the robot stand up.
+
+In the terminal, change directory to the location where the repository is cloned.
+
+### Select Branch
+
+If you are working within an existing project, then you need to find out which branch the work is on. To list the branches, use the command `git branch -r`. When you have found the branch you want to push to, swap to it using the following command:
+
+```sh
+git checkout <branch name>
+```
+
+Where `<branch name>` corresponds to the branch you found in the last step.
+
+If you are not working on an existing project, you need to make a branch to commit your changes to. The format for branch names is `surname/project_name`. So for this example it could be `citizen/stand_up`.
```

Comment:
```suggestion
If you are not working on an existing project, you need to make a branch to commit your changes to. The format for branch names is `your_surname/brief_description_of_project`. So if your surname is Hamiltons, and your project is making the robot stand up, a good branch name would be `hamiltons/stand_up`.
```

Reply:
Are we usually using real people's names in examples in NUbook?

Reply:
Yep :P

Reply:
I dunno, it just seemed kinda like, who on Earth would be named Hamiltons anyway??? stupid name :laughing: